### ISSUE TITLE
max kernels breached yields 402 -- Payment required ... but i ain't payin' !

### ISSUE THREAD

**Issue by rwhorman** [jupyter/kernel_gateway#158]
Officially, [w3](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html) says:

```
10.4.3 402 Payment Required

This code is reserved for future use.
```

Seems like a tad harsh of a punishment for breaching this max ;-) Not to mention, this really isn't officially a status code in play at this time.

So how about this as an alternative:

```
10.4.4 403 Forbidden

The server understood the request, but is refusing to fulfill it. Authorization will not help and the request SHOULD NOT be repeated. If the request method was not HEAD and the server wishes to make public why the request has not been fulfilled, it SHOULD describe the reason for the refusal in the entity. If the server does not wish to make this information available to the client, the status code 404 (Not Found) can be used instead.
```

This seems appropriate because, indeed, kernel-gateway cannot fulfill the request. Further, the response can include the specific reason. This will make it crystal clear to the caller of the API.

**Comment by parente** [comment]
The reasoning for 402 came from https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_Error

> 402 Payment Required
Reserved for future use. The original intention was that this code might be used as part of some form of digital cash or micropayment scheme, but that has not happened, and this code is not usually used. Google Developers API uses this status if a particular developer has exceeded the daily limit on requests.

Basically, copy what Google does. But, I don't really have a strong opinion either way. If it does indeed change, it'll have to go in 0.6 because it's an API change.

**Comment by rwhorman** [comment]
Words like "this code is not usually used" and "reserved for future use" strongly suggest to stay away. That being as it is, "payment required" is strange for open source software in general, let alone the unknowability of the use cases in which kernel-gateway could be utilized and what this message means contextually. 

To me it goes like this ...
gimme a kernel ... thx
gimme a kernel ... thx
gimme a kernal ... 403 -- nope, refusing to do so, there are no more kernels for you
